bundles/io/org.openhab.io.myopenhab/src/main/java/org/openhab/io/myopenhab/internal/MyOHClient.java

```diff
@@ -0,0 +1,559 @@
+	private HashMap<Integer, MyOHExchange> mRunningRequests;
+	// Active connection flag
+	private boolean mIsConnected;
+	// Version of openHAB
+	private String mOpenHABVersion;
+	// Socket.io
+	private Socket mSocket;
+	// MyOHClient listener
+	private MyOHClientListener mListener;
+
+	/**
+	 * Constructor of MyOHClient
+	 * 
+	 * @param uuid openHAB's UUID to connect to my.openHAB
+	 * @param secret openHAB's Secret to connect to my.openHAB
+	 * 
+	 */
+
+	public MyOHClient(String uuid, String secret) {
+		mUUID = uuid;
+		mSecret = secret;
+		mRunningRequests = new HashMap<Integer, MyOHExchange>();
+		mJettyClient = new HttpClient();
+		mJettyClient.setConnectorType(HttpClient.CONNECTOR_SELECT_CHANNEL);
+		mJettyClient.setMaxConnectionsPerAddress(200);
+		mJettyClient.setTimeout(30000);
+		try {
+			mJettyClient.start();
+		} catch (Exception e) {
+			logger.error("Error starting JettyClient: {}", e.getMessage());
+		}
+	}
+	
+	/**
+	 * Connect to my.openHAB
+	 */
+	
+	public void connect() {
+		try {
+			mSocket = IO.socket(mMyOHBaseUrl);
+			mSocket.io().on(Manager.EVENT_TRANSPORT, 
+				new Emitter.Listener() {
+					@Override
+					public void call(Object... args) {
+						logger.debug("Manager.EVENT_TRANSPORT");
+						Transport transport = (Transport)args[0];
+						transport.on(Transport.EVENT_REQUEST_HEADERS, new Emitter.Listener() {
+							@Override
+							public void call(Object... args) {
+								logger.debug("Transport.EVENT_REQUEST_HEADERS");
+								Map<String, String> headers = (Map<String, String>) args[0];
+								headers.put("uuid", mUUID);
+								headers.put("secret", mSecret);
+								headers.put("openhabversion", mOpenHABVersion);
+								headers.put("myohversion", MyOpenHABServiceImpl.myohVersion);
+							}
+						});
+					}
+				}
+			);
+			mSocket.on(Socket.EVENT_CONNECT, new Emitter.Listener() {
+				@Override
+				public void call(Object... args) {
+					logger.debug("Socket.IO connected");
+					mIsConnected = true;
+					onConnect();
+				}
+			}).on(Socket.EVENT_DISCONNECT, new Emitter.Listener() {
+				@Override
+				public void call(Object... args) {
+					logger.debug("Socket.IO disconnected");
+					mIsConnected = false;
+					onDisconnect();
+				}
+			}).on(Socket.EVENT_ERROR, new Emitter.Listener() {
+								@Override
+				public void call(Object... args) {
+					logger.error("Socket.IO error: " + args[0]);
+				}
+			}).on("request", new Emitter.Listener() {
+				@Override
+				public void call(Object... args) {
+					onEvent("request", (JSONObject)args[0]);
+				}
+			}).on("cancel", new Emitter.Listener() {
+				@Override
+				public void call(Object... args) {
+					onEvent("cancel", (JSONObject)args[0]);
+				}
+			}).on("command", new Emitter.Listener() {
+				
+				@Override
+				public void call(Object... args) {
+					onEvent("command", (JSONObject)args[0]);
+				}
+			});
+			mSocket.connect();
+		} catch (URISyntaxException e) {
```

Comment:
what exactly throws the USE? Please try to keep the try-catch-block as small as possible